torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -470,6 +470,7 @@ ProcessGroupNCCL::WorkNCCL::WorkNCCL(
     const std::optional<std::vector<at::Tensor>>& inputs,
     bool desyncDebug,
     bool enableTiming,
+    bool cudaEventCacheEnabled,
```

Comment:
nit: maybe we can make `WorkNCCL` a friend of `ProcessGroupNCCL` so that it can access `ProcessGroupNCCL. cudaEventCacheEnabled_` and we don't have to pass every flag to the constructor of `WorkNCCL`?